packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/README.md

```diff
@@ -0,0 +1,377 @@
+# Technical documentation
+
+Here we will cover how the `<Permissions />` component works.
+This component is in charge of managing the permissions of the admin panel, it has 4 children one child for each tab.
+The collection types & single types tabs uses the same component: `<ContentTypes>` similarly, the settings & plugins tab uses the `<Plugins>` component.
+
+The UI uses the layout received from the back-end in order to build the UI.
+
+## Layout shape
+
+```js
+const layout = {
+  conditions: [{ id: 'string', displayName: 'string', category: 'string'}], // Array of conditions that can be applied on a permission
+  sections: {
+    plugins: [
+      { displayName: 'string', action: 'string', subCategory: 'string', plugin: 'string'}
+    ],
+    settings: [], // Same shape as the plugins
+    collectionTypes: {
+      subjects: [ // Array of subjects
+        {
+          uid: 'string', // Collection type uid,
+          label: 'string', // Collection type label,
+          properties: [
+            {
+              label: 'string', // Property label ex: Fields,
+              value: 'string', // Value of the property ex: fields,
+              children: [ // This corresponds to the fields that we will display
+                {
+                  label: 'string',
+                  value: 'string',
+                  required: 'boolean', // This key is optional,
+                  children: [], // This key is optional, if it exists it means that a field has nested fields
+                }
+              ]
+             },
+          ]
+        }
+      ],
+      actions: [ // Array of actions
+        {
+          label: 'string', // Label of the action ex: Create,
+          actionId: 'string', // id of the action ex: content-manager.explorer.create
+          subjects: [<string>], // Array of subjects (collection type uid) on which the action can be applied
+          applyToProperties: [<string>] // Array of properties (ex: fields or locales) on which an action can be applied
+        }
+      ]
+    }
+  }
+}
+```
+
+---
+
+## Concepts
+
+- Checkbox: has only 1 `checked=true` or `checked=false`.
+
+- Parent checkbox: A **parent checkbox** is a checkbox which the state value depends on the state of its children ones. It means that we cannot directly access its value from the `modifiedData` object.
+  Such checkbox can have multiple states:
+  - someChecked: `true` or `false`,
+  - checked: `true` of `false`
+    > Both states are coupled: if `someChecked=true` then `checked=false`.
+    > In terms of user's interaction when cliking a parent checkbox is will toggle the value of its children.
+
+Ex: given the following data:
+
+```js
+const modifiedData = {
```

Comment:
What do you think of naming this like `modifiedPermissions`?

Reply:
In the admin panel, we always deal with `modifiedData` in a form so I want to keep it consistent.